src/enumerable/enumerable_gather.jl

```diff
@@ -0,0 +1,75 @@
+struct EnumerableGather{T,S,F,I} <: Enumerable
+    source::S
+    fields::F
+    indexFields::I
+    key::Symbol
+    value::Symbol
+end
+
+struct Not{T}
+    val::T
+end
+
+function gather(source::Enumerable, args...; key::Symbol = :key, value::Symbol = :value)
+    T = eltype(source)
+    fields = fieldnames(T)
+    F = typeof(fields)
+    if args != (false,)
```

Comment:
Instead of passing a sentinel value of ``false`` here, can't we just check whether the length of ``args`` is 0 for the case where no field names are passed?

Reply:
I've changed this to `length(args)`.